File: ChatSupport/src/main/java/com/vantagecircle/chatapp/core/AuthHandler.java
package com.vantagecircle.chatapp.core;

import android.support.annotation.NonNull;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.vantagecircle.chatapp.core.interfaceC.ResultInterface;

/**
 * Created by bapidas on 26/07/17.
 */

public class AuthHandler {
    private FirebaseAuth firebaseAuth;

    public void setFirebaseAuth(FirebaseAuth firebaseAuth) {
        this.firebaseAuth = firebaseAuth;
    }

    public void performLogin(String email, String password, final ResultInterface resultInterface) {
        firebaseAuth.signInWithEmailAndPassword(email, password)
                .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
                    @Override
                    public void onComplete(@NonNull Task<AuthResult> task) {
                        if (task.isSuccessful()) {
                            resultInterface.onSuccess(task.toString());
                        }
                    }
                })
                .addOnFailureListener(new OnFailureListener() {
                    @Override
                    public void onFailure(@NonNull Exception e) {
                        resultInterface.onFail(e.getMessage());
                    }
                });
    }

    public void performSignup(String email, String password, final ResultInterface resultInterface) {
        firebaseAuth.createUserWithEmailAndPassword(email, password)
                .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
                    @Override
                    public void onComplete(@NonNull Task<AuthResult> task) {
                        if (task.isSuccessful()) {
                            resultInterface.onSuccess(task.toString());
                        }
                    }
                })
                .addOnFailureListener(new OnFailureListener() {
                    @Override
                    public void onFailure(@NonNull Exception e) {
                        resultInterface.onFail(e.getMessage());
                    }
                });
    }
}

File: app/src/main/java/com/bapidas/chattingapp/data/core/AuthHandler.kt
package com.bapidas.chattingapp.data.core

import com.bapidas.chattingapp.data.core.callbacks.ResultInterface
import com.google.firebase.auth.FirebaseAuth

/**
 * Created by bapidas on 26/07/17.
 */
class AuthHandler {
    lateinit var firebaseAuth: FirebaseAuth

    fun performLogin(email: String, password: String, resultInterface: ResultInterface) {
        firebaseAuth.signInWithEmailAndPassword(email, password)
                .addOnCompleteListener { task ->
                    if (task.isSuccessful) {
                        resultInterface.onSuccess(task.toString())
                    }
                }
                .addOnFailureListener { e -> resultInterface.onFail(e.message.orEmpty()) }
    }

    fun performSignup(email: String, password: String, resultInterface: ResultInterface) {
        firebaseAuth.createUserWithEmailAndPassword(email, password)
                .addOnCompleteListener { task ->
                    if (task.isSuccessful) {
                        resultInterface.onSuccess(task.toString())
                    }
                }
                .addOnFailureListener { e -> resultInterface.onFail(e.message.orEmpty()) }
    }
}